src/reason-parser/reason_pprint_ast.ml

```diff
@@ -1822,25 +1822,28 @@ let is_direct_pattern x = x.ppat_attributes = [] && match x.ppat_desc with
   | Ppat_construct ( {txt= Lident"()"}, None) -> true
   | _ -> false
 
-let isJSXComponent loc args =
-  let hasLabelledChildrenLiteral = List.exists (function
-    | (Labelled "children", _) -> true
-    | _ -> false
-  ) args in
-  let rec hasSingleNonLabelledUnitAndIsAtTheEnd l = match l with
-  | [] -> false
-  | (Nolabel, {pexp_desc = Pexp_construct ({txt = Lident "()"}, _)}) :: [] -> true
-  | (Nolabel, _) :: rest -> false
-  | _ :: rest -> hasSingleNonLabelledUnitAndIsAtTheEnd rest
-  in
-  if hasLabelledChildrenLiteral && hasSingleNonLabelledUnitAndIsAtTheEnd args then
-    let moduleNameList = List.rev (List.tl (List.rev (Longident.flatten loc.txt))) in
-    if List.length moduleNameList > 0 && Longident.last loc.txt = "createElement" then
-       true
-    else false
-  else
-    false
-
+let isJSXComponent expr =
+  match expr with
+  | ({pexp_desc= Pexp_apply ({pexp_desc=Pexp_ident _}, args); pexp_attributes}) ->
+    let {jsxAttrs} = partitionAttributes pexp_attributes in
+    let hasLabelledChildrenLiteral = List.exists (function
+      | (Labelled "children", _) -> true
+      | _ -> false
+    ) args in
+    let rec hasSingleNonLabelledUnitAndIsAtTheEnd l = match l with
+    | [] -> false
+    | (Nolabel, {pexp_desc = Pexp_construct ({txt = Lident "()"}, _)}) :: [] -> true
+    | (Nolabel, _) :: rest -> false
+    | _ :: rest -> hasSingleNonLabelledUnitAndIsAtTheEnd rest
+    in
+    if List.length jsxAttrs > 0
```

Comment:
Doing `jsxAttrs = []` is O(1) instead of O(n). Or just match on a 3-tuple

Reply:
Thanks! Should have thought about this...